src/variables/create.jl

```diff
@@ -0,0 +1,156 @@
+export compute_variables_indices
+
+# TODO: Allow changing table names to make unit tests possible
+# The signature should be something like `...(connection; assets_data="t_assets_data", ...)`
+function compute_variables_indices(connection, dataframes)
+    variables = Dict(
+        :flow => TulipaVariable(dataframes[:flows]),
+        :units_on => TulipaVariable(dataframes[:units_on]),
+        :storage_level_intra_rp => TulipaVariable(dataframes[:storage_level_intra_rp]),
+        :storage_level_inter_rp => TulipaVariable(dataframes[:storage_level_inter_rp]),
+        :is_charging => TulipaVariable(dataframes[:lowest_in_out]),
+    )
+
+    variables[:flows_investment] = TulipaVariable(DuckDB.query(
+        connection,
+        "SELECT
+            from_asset,
+            to_asset,
+            year,
+            investment_integer,
+        FROM flows_data
+        WHERE
+            year=commission_year
+            AND investable=true",
+    ) |> DataFrame)
+    dataframes[:flows_investment] = variables[:flows_investment].indices
+
+    # TODO: Verify that year=commission_year is the correct approach. Alternative is creating a table where (name, year) is the key
+    variables[:assets_investment] = TulipaVariable(DuckDB.query(
+        connection,
+        "SELECT
+            name,
+            year,
+            investment_integer,
+        FROM assets_data
+        WHERE
+            year=commission_year
+            AND investable=true",
+    ) |> DataFrame)
+    dataframes[:assets_investment] = variables[:assets_investment].indices
+
+    variables[:assets_decommission_simple_method] = TulipaVariable(
+        DuckDB.query(
+            connection,
+            "SELECT
+                assets_data.name,
+                assets_data.year,
+                assets_data.investment_integer,
+            FROM assets_data
+            LEFT JOIN graph_assets_data
+                ON assets_data.name=graph_assets_data.name
+            WHERE
+                year=commission_year
+                AND investment_method='simple'",
+        ) |> DataFrame,
+    )
+
+    # For the compact method, not all combinations of (assets, year, commission_year) exist
+    # In particular, if the commission years are also milestone years where the asset is investable. 
+    # which means that we use the UNION to add them
+    # We only filter afterwards to increase readability, but it would be technically more
+    # efficient to filter for each inner query before computing the union
+    # NOTE: To add another column from assets_data, it has to be added on the two inner SELECT
+    # statements (not the year_data one)
+    variables[:assets_decommission_compact_method] = TulipaVariable(
+        DuckDB.query(
```

Comment:
Thanks @abelsiqueira, @gnawin and I reviewed this query and it is almost there. So, two things:

1. We tested it in the multi-year investment data, and it works. However, we want to cover another edge case outside that use case (so, in issue #684 I added that to the list to improve it). If we change the input to have the asset = ccgt in year = 2030 and commission_year = 2030 with investable = false, then it should not appear in the result of this query. As it is, it seems. Ni and I tried to fix it but couldn't (sorry about that). If we cover that case, then this change is okay.
2. For the indices of the decommission variables, we don't need the investable column (we need the investment integer for creating the variable). So, we were wondering, why do you keep it? Maybe we are missing something. If we don't need it, then we should take it out.

Reply:
1. Can we add that to the actual tests? Otherwise there is guarantee that it works as expected. 
   I also don't see the relation between this new data row and the compact formulation. There are other rows that result in `investable = false` in the query, so `investable` is not taken into account

2. I was just using for checking that things propagate